### DISCUSSION TITLE
Font types and Cumulative Layout Shift

### DISCUSSION THREAD

**Discussion by andrecasal** [epicweb-dev/epic-stack#175] [Q&A]
Three questions:
1. [Google Fonts](https://fonts.google.com/specimen/Nunito+Sans) only has TTF versions for Nunito Sans. Do you convert these into woff and woff2 or do you download them from somewhere else?
2. As a best practice, shouldn't we be using Nunito Sans's [variable font](https://fonts.google.com/knowledge/introducing_type/introducing_variable_fonts)? The complete set of woff/woff2 is 838 kB and the variable font is 569 kB. We'd save 269 kB overall, but the first load would still be penalized as it only needs 105.7 kB:
![Screenshot 2023-06-14 at 13 29 30](https://github.com/epicweb-dev/epic-stack/assets/22457463/a6e03837-6703-426d-8569-f2260403257a)
3. Is there a fontpie alternative that works for variable fonts?

**Comment by kentcdodds** [comment]
I can't remember where I got the woff and woff2 versions of the font, but I do remember I had to look around the web for them.

I'd be happy to accept a PR that allows us to use that font but smaller.

I don't know whether there's a fontpie alternative that supports variable fonts. @xHomu do you know?

**Reply by xHomu** [reply]
I use fontsquirrel to generate optimized subsets, it'll output .woff and .woff2 files https://www.fontsquirrel.com/tools/webfont-generator

**Reply by xHomu** [reply]
I'm of the opinion that we don't need to preload fonts at all. With font swap, we made sure the display text is rendered before the fonts streamed in. With font metric overrides, we ensured that CLS would not be a problem.

**Reply by kentcdodds** [reply]
By that you mean we can remove these lines? https://github.com/epicweb-dev/epic-stack/blob/4ab28e37b2ea20c75fea3558e6637ef102bd3207/app/root.tsx#L40-L43

**Reply by xHomu** [reply]
Ops, thought Epic Stack was preloading fonts files in root too. Ignore that! 

Preload css, don't preload fonts!
